src/hct_mis_api/apps/payment/celery_tasks.py

```diff
@@ -632,3 +635,99 @@ def periodic_sync_payment_gateway_delivery_mechanisms(self: Any) -> None:
     except Exception as e:
         logger.exception(e)
         raise self.retry(exc=e)
+
+
+@app.task(bind=True, queue="priority", default_retry_delay=60, max_retries=3)
+@log_start_and_end
+@sentry_tags
+def payment_plan_apply_steficon_hh_selection(self: Any, payment_plan_id: str, engine_rule_id: str) -> None:
+    from hct_mis_api.apps.payment.models import Payment, PaymentPlan
+    from hct_mis_api.apps.steficon.models import Rule, RuleCommit
+
+    payment_plan = get_object_or_404(PaymentPlan, id=payment_plan_id)
+    set_sentry_business_area_tag(payment_plan.business_area.name)
+    engine_rule = get_object_or_404(Rule, id=engine_rule_id)
+    rule: Optional["RuleCommit"] = engine_rule.latest
+    if rule and rule.id != payment_plan.steficon_rule_targeting_id:
+        payment_plan.steficon_rule_targeting = rule
+        payment_plan.save(update_fields=["steficon_rule_targeting"])
+    try:
+        payment_plan.status = PaymentPlan.Status.TP_STEFICON_RUN
+        payment_plan.steficon_targeting_applied_date = timezone.now()
+        payment_plan.save(update_fields=["status", "steficon_targeting_applied_date"])
+        updates = []
+        with transaction.atomic():
+            payment: Payment
+            for payment in payment_plan.payment_items.all():
+                result = rule.execute(
+                    {
+                        "household": payment.household,
+                        "payment_plan": payment_plan,
```

Comment:
We need to remember to handle "payment_plan" keyword in rules code

Reply:
yes we have to notice that and I will add comment about that in the ticket